rmw_fastrtps_shared_cpp/src/custom_subscriber_info.cpp

```diff
@@ -104,6 +124,15 @@ bool SubListener::takeNextEvent(rmw_event_type_t event_type, void * event_info)
         deadline_changes_.store(false, std::memory_order_relaxed);
       }
       break;
+    case RMW_EVENT_MESSAGE_LOST:
+      {
+        auto rmw_data = static_cast<rmw_message_lost_status_t *>(event_info);
+        rmw_data->total_count = sample_lost_status_.total_count;
+        rmw_data->total_count_change = sample_lost_status_.total_count_change;
+        sample_lost_status_.total_count_change = 0;
+        sample_lost_changes_.store(false, std::memory_order_relaxed);
```

Comment:
So, can `total_count_change` not be used as a proxy for `sample_lost_changes_`? Like if `total_count_change` != 0, then `sample_lost_changes_` will always be `true`, and the opposite as well, if it is == 0, then `sample_lost_changes_` will always be `false`, right?

In which case, it might make sense to wrap the entire thing as `std::atomic<eprosima::fastdds::dds::SampleLostStatus>`? Depending on the size of the fields in that struct, and the processor it may still be truly atomic, otherwise it will wrap it in a mutex.